src/generateDiff.js

```diff
@@ -11,30 +18,64 @@ const diff = require('diff');
  */
 function generateDiff(before, after) {
   const changes = diff.diffLines(before, after);
-  const diffLines = [];
-  const preceedingContext = [];
-  for (const { added, removed, value } of changes) {
-    const lines = value.split('\n');
-    // remove trailing newline
-    lines.pop();
-    if (added || removed) {
-      if (preceedingContext.length) {
-        diffLines.push(...preceedingContext);
-        preceedingContext.splice(0, preceedingContext.length);
-      }
-      diffLines.push(...lines.map((line) => `${added ? '+' : '-'}${line}`));
-    } else {
-      // If a changed line has been included already, add up to 2 lines of context
-      if (diffLines.length) {
-        diffLines.push(...lines.slice(0, 2).map((line) => ` ${line}`));
-        lines.splice(0, 2);
+  // `diffLines` will always return at least one change
```

Comment:
Will it always return at least one _change_ or one _value_?

Reply:
Ah, right, I see that I was using the term 'change' to mean different things in this file :thinking:
I can re-word this to be a bit more clear. Maybe I can call this a `diffResult` rather than a "change"? As it includes both additions, removals, and sections where no change occured.

Reply:
The library calls these "change objects", where each "change object" might be a removal, an addition, or a section that hadn't changed.